feat(network): add conditional VPC endpoints creation

axel-ippon opened the pull request:
## Purpose of this PR

Introduces create_vpc_endpoints variable to make VPC endpoint resources optional, allowing users to choose between VPC endpoints and NAT gateway approaches for private subnet internet access.

All VPC endpoint-related resources now use count-based conditional creation, improving infrastructure flexibility and cost optimization options.

## Reference issues or tasks 

[307](https://github.com/ippontech/iroco2/issues/307)

axel-ippon committed:
feat(network): add conditional VPC endpoints creation

Introduces create_vpc_endpoints variable to make VPC endpoint resources optional, allowing users to choose between VPC endpoints and NAT gateway approaches for private subnet internet access.

All VPC endpoint-related resources now use count-based conditional creation, improving infrastructure flexibility and cost optimization options.

axel-ippon committed:
refactor(network): switch from VPC endpoints to NAT gateway egress

Conditionally creates allow-all egress rules when VPC endpoints are disabled
and restricts prefix list rules to only when VPC endpoints are enabled.
This change supports the migration from VPC endpoints to NAT gateway
for internet access while maintaining security group rule consistency.

Donovan1905 commented on main.tf:
```diff
@@ -5,14 +5,15 @@ module "network" {
+  zone_name            = var.zone_name
+  enable_nat_gateway   = var.enable_nat_gateway
+  single_nat_gateway   = var.single_nat_gateway
+  create_vpc_endpoints = var.create_vpc_endpoints
```
It could be great to be sure that if `create_vpc_endpoints` is `false`, `enable_nat_gateway` is `true`, otherwise, the ECS task won't be able to pull the backend docker image.

I don´t know if, I am good on this. What do you think ?

axel-ippon replied:
Yep we can force that I agree, it's done in this [commit](https://github.com/ippontech/iroco2-terraform-modules/pull/13/commits/682fd584851244f83c9397e1bc2d3071b1113d14#diff-dc46acf24afd63ef8c556b77c126ccc6e578bc87e3aa09a931f33d9bf2532fbbR14) 🙂

axel-ippon committed:
refactor(network): switch from VPC endpoints to NAT gateway by default

Changes network configuration to use NAT gateway as the default networking solution
instead of VPC endpoints. Removes manual NAT gateway variables and implements
automatic conditional logic where NAT gateway is enabled when VPC endpoints are
disabled.
